lib/ddtrace/distributed_headers.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/span'
+require 'ddtrace/ext/distributed'
+
+module Datadog
+  # DistributedHeaders provides easy access and validation to headers
+  class DistributedHeaders
+    include Ext::DistributedTracing
+
+    def initialize(env)
+      @env = env
+    end
+
+    def valid?
+      # Sampling priority is optional.
+      trace_id && parent_id
+    end
+
+    def trace_id
+      value = header(HTTP_HEADER_TRACE_ID).to_i
+      return if value <= 0 || value >= Span::MAX_ID
+      value
+    end
+
+    def parent_id
+      value = header(HTTP_HEADER_PARENT_ID).to_i
+      return if value <= 0 || value >= Span::MAX_ID
+      value
+    end
+
+    def sampling_priority
+      hdr = header(HTTP_HEADER_SAMPLING_PRIORITY)
+      # It's important to make a difference between no header,
+      # and a header defined to zero.
```

Comment:
This is required because some hosts might have priority sampling disabled and/or have outdated libraries so they are not propagating the information. In this case, we want to just ignore it and not set it.

Reply:
👍